functions/levante-admin/src/upsertAdministration.ts

```diff
@@ -363,15 +638,98 @@ export const upsertAdministrationHandler = async (
           );
         }
       }
-      logger.info(`Successfully prepared administration ${operationType}`, {
+      logger.info("Successfully prepared administration", {
         administrationId: administrationDocRef.id,
+        operationType: administrationId ? "update" : "create",
       });
       return administrationDocRef.id; // Return the ID
     }); // End Transaction
 
     logger.info("Finished administration upsert transaction", {
       administrationId: newAdministrationId,
     });
+
+    // Sync assignments synchronously for immediate visibility
+    try {
+      const administrationDocRef = db
+        .collection("administrations")
+        .doc(newAdministrationId);
+      const administrationDoc = await administrationDocRef.get();
+
+      if (!administrationDoc.exists) {
+        logger.warn(
+          `Administration ${newAdministrationId} not found after creation. Skipping sync.`
+        );
+        return { status: "ok", administrationId: newAdministrationId };
+      }
+
+      const administrationData = administrationDoc.data() as IAdministration;
+      const isNewAdministration = !administrationId;
+      const creatorUid = administrationData?.createdBy;
+
+      if (isNewAdministration) {
+        if (!creatorUid) {
+          logger.error(
+            `Cannot sync assignments: administration ${newAdministrationId} has no createdBy field`,
+            { administrationId: newAdministrationId }
+          );
+          throw new HttpsError(
+            "internal",
+            "Administration document is missing creator information"
+          );
```

Comment:
This shouldn't be possible because the creatorId will always be on the auth token when we first create the administration document, so it could never be missing it.